### PR TITLE
Integration db

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/BuildingRegistry.Projections.Integration/Building/LatestItem/BuildingLatestItem.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            get => Instant.FromDateTimeOffset(VersionTimestampAsDateTimeOffset);
+            set
+            {
+                VersionTimestampAsDateTimeOffset = value.ToDateTimeOffset();
+                VersionAsString = new Rfc3339SerializableDateTimeOffset(value.ToBelgianDateTimeOffset()).ToString();
+            }
+        }
+
+        public string CreatedOnAsString { get; set; }
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Integration/Building/LatestItem/BuildingLatestItem.cs:40]: Not needed here, only in version.

File: src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitLatestItem.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public sealed class BuildingUnitLatestItem
+    {
+        public const string VersionTimestampBackingPropertyName = nameof(VersionTimestampAsDateTimeOffset);
+        public const string CreatedTimestampBackingPropertyName = nameof(CreatedOnTimestampAsDateTimeOffset);
+
+        public int BuildingUnitPersistentLocalId { get; set; }
+        public int BuildingPersistentLocalId { get; set; }
+        public string Status { get; set; }
+        public string OsloStatus { get; set; }
+        public string Function { get; set; }
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitLatestItem.cs:21]: OsloFunction + Function

File: src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitLatestItem.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        public string GeometryMethod { get; set; }
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitLatestItem.cs:22]: OsloGeometryMethod + GeometryMethod

File: src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitLatestItem.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            get => Instant.FromDateTimeOffset(VersionTimestampAsDateTimeOffset);
+            set
+            {
+                VersionTimestampAsDateTimeOffset = value.ToDateTimeOffset();
+                VersionAsString = new Rfc3339SerializableDateTimeOffset(value.ToBelgianDateTimeOffset()).ToString();
+            }
+        }
+
+        public string CreatedOnAsString { get; set; }
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitLatestItem.cs:42]: Not needed

File: src/BuildingRegistry.Projections.Integration/IntegrationContext.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        public override string ProjectionStateSchema => Schema.Integration;
+
+        public DbSet<BuildingLatestItem> BuildingLatestItems => Set<BuildingLatestItem>();
+
+        public DbSet<BuildingUnitLatestItem> BuildingUnitLatestItems => Set<BuildingUnitLatestItem>();
+        public DbSet<BuildingUnitAddress> BuildingUnitAddresses => Set<BuildingUnitAddress>();
+
+        private DbSet<MunicipalityGeometry> MunicipalityGeometries => Set<MunicipalityGeometry>();
+
+        public async Task<string?> FindNiscode(Geometry sysGeometry, CancellationToken ct)
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Integration/IntegrationContext.cs:24]: `FindNisCodeBy`?

Comment (ArneD) [reply src/BuildingRegistry.Projections.Integration/IntegrationContext.cs:24]: Or `FindMostIntersectingNisCodeBy` ?

File: src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitAddress.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        {
+            const string tableName = "building_unit_addresses";
+
+            builder
+                .ToTable(tableName, Schema.Integration) // to schema per type
+                .HasKey(x => new { x.BuildingUnitPersistentLocalId, x.AddressPersistentLocalId });
+
+            builder.Property(x => x.BuildingUnitPersistentLocalId).HasColumnName("building_unit_persistent_local_id");
+            builder.Property(x => x.AddressPersistentLocalId).HasColumnName("address_persistent_local_id");
+        }
```
Comment (ArneD) [inline src/BuildingRegistry.Projections.Integration/BuildingUnit/LatestItem/BuildingUnitAddress.cs:28]: Add index on separate properties

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/BuildingRegistry.Projections.Integration/Building/LatestItem/BuildingLatestItem.cs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            builder.Property(x => x.Geometry).HasColumnName("geometry");
+            builder.Property(x => x.NisCode).HasColumnName("nis_code");
+            builder.Property(x => x.IsRemoved).HasColumnName("is_removed");
+            builder.Property(x => x.PuriId).HasColumnName("puri_id");
+            builder.Property(x => x.Namespace).HasColumnName("namespace");
+            builder.Property(x => x.VersionAsString).HasColumnName("version_as_string");
+            builder.Property(BuildingLatestItem.VersionTimestampBackingPropertyName).HasColumnName("version_timestamp");
+            builder.Property(x => x.CreatedOnAsString).HasColumnName("created_on_as_string");
+            builder.Property(BuildingLatestItem.CreatedTimestampBackingPropertyName).HasColumnName("created_on_timestamp");
+
```
Comment (emalfroy) [inline src/BuildingRegistry.Projections.Integration/Building/LatestItem/BuildingLatestItem.cs:79]: Index on timestamp

**Review by emalfroy — APPROVED** [review]

**Review by ArneD — APPROVED** [review]

**Pull request merged**